src/components/EventCalender/Monthly/EventCalender.tsx

```
@@ -294,26 +285,23 @@ const Calendar: React.FC<InterfaceCalendarProps> = ({
                   <p className={styles.no_events_message}>
                     No events available
                   </p>
                 )}
               </div>
-              {Array.isArray(allDayEventsList) && (
+              {Array.isArray(allDayEventsList) && shouldShowViewMore && (
                 <button
                   className={styles.btn__more}
                   onClick={handleExpandClick}
+                  data-testid="view-more-button"
                 >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)